lib/pbench/server/filetree.py

```diff
@@ -322,30 +322,27 @@ def get_metadata(self) -> JSONOBJECT:
 
     @staticmethod
     def subprocess_run(
-        command: str, curr_dir: Path, exception: FiletreeError, ctx: Path
+        command: str, working_dir: Path, exception: FiletreeError, ctx: Path
```

Comment:
The type hint still shows an *instance* of `FiletreeError`; it's really a *subclass* of `FiletreeError` (or, as I suggested elsewhere, a more specific class extended by your two intended exceptions). So `type[FiletreeError]` would be more appropriate.

Also, in the `Args` list, specify that it's a class name.

Reply:
missed it.

Reply:
Oh, and "beware" ... this syntax is a Python 3.9-ism that will be rejected by Python 3.6 (`type object is not subscriptable`), so if you're unit testing or deploying with 3.6, you need to upgrade to 3.9!

(I got caught on that today after putting a similar type hint in some new code, and discovering that my test server is still defaulting to Python 3.6.)

Reply:
So, perhaps (at least for the interim) the type hint for `exception` should be just `type` (with no specialization)?  It's a bit generic, but it distinguishes the value from an _instance_ of an exception.

Reply:
We made the decision to target 3.9 for the server, and I think that's a good decision. I *like* being able to use standard type names as generics (as here). It was easy enough to change the default on my server to 3.9. (Though the spec file `%post` started tripping over `psycopg2` and I changed to `psycopg2-binary`, which I might just retain, until we find an RPM... which, after all, will presumably be analogous to the `-binary` anyway.)